### PR TITLE
more/better testing functions and added support for files

### PR DESCRIPTION
**Pull request by bravosierra99**
Mostly added some more helper functions for testing and added the peek ability we talked about previously.  Also made it so you can actually read from attachments that the bot sends.

### CODE REVIEW

File: discord/ext/test/backend.pyi [hunk 1]
```diff
@@ -82,7 +82,7 @@ def make_member(user: discord.User, guild: discord.Guild, nick: str = ..., roles
 
 def update_member(member: discord.Member, nick: str = ..., roles: List[discord.Role] = ...) -> discord.Member: ...
 
-def make_message(content: str, author: Union[discord.User, discord.Member], channel: AnyChannel, id_num: int = ...) -> discord.Message: ...
+def make_message(content: str, author: Union[discord.User, discord.Member], channel: AnyChannel, attachments : List[Attachment] id_num: int = ...) -> discord.Message: ...
```
Comment (CraftSpider) [inline discord/ext/test/backend.pyi:85]: This PR doesn't alter backend.py, so this change would be incorrect

File: discord/ext/test/runner.pyi [hunk 2]
```diff
@@ -19,9 +19,13 @@ error_queue: asyncio.Queue
 
 async def run_all_events() -> None: ...
 
-def verify_message(text: str = ..., equals: bool = ...) -> None: ...
+def verify_message(text: str = ..., equals: bool = ..., contains: bool = ..., peek: bool = ...) -> None: ...
 
-def verify_embed(embed: discord.Embed = ..., allow_text: bool = ..., equals: bool = ...) -> None: ...
+def get_message(peek : bool =False) -> discord.Message: ...
```
Comment (CraftSpider) [inline discord/ext/test/runner.pyi:24]:
```suggestion
def get_message(peek: bool = ...) -> discord.Message: ...
```

File: discord/ext/test/runner.pyi [hunk 3]
```diff
@@ -19,9 +19,13 @@ error_queue: asyncio.Queue
...
+
+def verify_embed(embed: discord.Embed = ..., allow_text: bool = ..., equals: bool = ..., peek: bool = ...) -> None: ...
+
+def get_message(peek : bool =False) -> discord.Embed: ...
```
Comment (CraftSpider) [inline discord/ext/test/runner.pyi:28]:
```suggestion
def get_message(peek: bool = ...) -> discord.Embed: ...
```

**Comment by CraftSpider** [comment]
Other than the changes above, I think this looks good for now. I may look into changing the queue to be something that allows peeking more easily later

**Comment by bravosierra99** [comment]
Absolutely. I’ll try and get to this today, if not tomorrow. Thanks!

**Comment by bravosierra99** [comment]
this is one of my first times doing a pull request to some else's repo, I've made the change as requested, do I need to do anything else?  It doesn't look like I need to resubmit the pull, though my repo is now behind yours so I'm not sure if I need to wait for this to go through or if I should update my master by pulling yours in.  I may take a look at that.  My apologies, just new to this.

**Comment by bravosierra99** [comment]
ok pulled in your v0.0.21, so that should make it easy to merge in yours I thinkk.

**Comment by CraftSpider** [comment]
Once travis passes, I'll merge it in

**Pull request merged**